src/rgw/rgw_data_sync.cc

```diff
@@ -1717,8 +1906,20 @@ class RGWDataSyncShardCR : public RGWCoroutine {
                                                error_marker, entry_timestamp), false);
               continue;
             }
-            tn->log(20, SSTR("handle error entry key=" << to_string(source_bs, gen) << " timestamp=" << entry_timestamp));
-            spawn(sync_single_entry(source_bs, gen, "", entry_timestamp, true), false);
+            if (!gen) {
+              // write all full sync obligations for the bucket to error repo
+              spawn(new RGWFullSyncErrorRepoCR(sc, source_bs, entry_timestamp, &error_result, tn), false);
+              if (error_result == 0) {
+                spawn(rgw::error_repo::remove_cr(sync_env->store->svc()->rados, error_repo,
+                                                error_marker, entry_timestamp), false);
```

Comment:
`spawn()` is asynchronous, so you can't expect `error_result` to be assigned immediately. this logic to call `rgw::error_repo::remove_cr()` should just happen at the end of `RGWFullSyncErrorRepoCR`. that way you don't need to share this `int error_result` between the two coroutines